packages/material-ui/src/Alert/Alert.js

```diff
@@ -1,147 +1,213 @@
+      styleProps.color === 'error' && {
+        color: getColor(theme.palette.error.main, 0.6),
+        backgroundColor: getBackgroundColor(theme.palette.error.main, 0.9),
+        '& $icon': {
+          color: theme.palette.error.main,
+        },
+      }),
     /* Styles applied to the root element if `variant="outlined"` and `color="success"`. */
-    outlinedSuccess: {
-      color: getColor(theme.palette.success.main, 0.6),
-      border: `1px solid ${theme.palette.success.main}`,
-      '& $icon': {
-        color: theme.palette.success.main,
-      },
-    },
+    ...(styleProps.variant === 'outlined' &&
+      styleProps.color === 'success' && {
+        color: getColor(theme.palette.success.main, 0.6),
+        border: `1px solid ${theme.palette.success.main}`,
+        '& $icon': {
+          color: theme.palette.success.main,
+        },
+      }),
     /* Styles applied to the root element if `variant="outlined"` and `color="info"`. */
-    outlinedInfo: {
-      color: getColor(theme.palette.info.main, 0.6),
-      border: `1px solid ${theme.palette.info.main}`,
-      '& $icon': {
-        color: theme.palette.info.main,
-      },
-    },
+    ...(styleProps.variant === 'outlined' &&
+      styleProps.color === 'info' && {
+        color: getColor(theme.palette.info.main, 0.6),
+        border: `1px solid ${theme.palette.info.main}`,
+        '& $icon': {
+          color: theme.palette.info.main,
+        },
+      }),
     /* Styles applied to the root element if `variant="outlined"` and `color="warning"`. */
-    outlinedWarning: {
-      color: getColor(theme.palette.warning.main, 0.6),
-      border: `1px solid ${theme.palette.warning.main}`,
-      '& $icon': {
-        color: theme.palette.warning.main,
-      },
-    },
+    ...(styleProps.variant === 'outlined' &&
+      styleProps.color === 'warning' && {
+        color: getColor(theme.palette.warning.main, 0.6),
+        border: `1px solid ${theme.palette.warning.main}`,
+        '& $icon': {
+          color: theme.palette.warning.main,
+        },
+      }),
     /* Styles applied to the root element if `variant="outlined"` and `color="error"`. */
-    outlinedError: {
-      color: getColor(theme.palette.error.main, 0.6),
-      border: `1px solid ${theme.palette.error.main}`,
-      '& $icon': {
-        color: theme.palette.error.main,
-      },
-    },
+    ...(styleProps.variant === 'outlined' &&
+      styleProps.color === 'error' && {
+        color: getColor(theme.palette.error.main, 0.6),
+        border: `1px solid ${theme.palette.error.main}`,
+        '& $icon': {
+          color: theme.palette.error.main,
+        },
+      }),
     /* Styles applied to the root element if `variant="filled"` and `color="success"`. */
-    filledSuccess: {
-      color: '#fff',
-      fontWeight: theme.typography.fontWeightMedium,
-      backgroundColor: theme.palette.success.main,
-    },
+    ...(styleProps.variant === 'filled' &&
```

Comment:
There are lots of duplications on these conditional styles. I would propose refactoring it to something like this:

```
...(styleProps.variant === 'filled' && {
  color: '#fff',
  fontWeight: theme.typography.fontWeightMedium,
  backgroundColor: theme.palette.[styleProps.color].main,
}),
```
With `  backgroundColor: theme.palette.[styleProps.color].main,` we are basically taking care of all colors, based on which one is used :)

Reply:
Hint: Try to do these for the over variants as well :)